### ISSUE TITLE
Stop Spring Bom from being applied twice

### ISSUE THREAD

**Issue by codeconsole** [grails/grails-core#14145]
Now that the Spring Bom is parent to grails-bom.
The following code needs to be evaluated:

https://github.com/grails/grails-gradle-plugin/blob/755b810a1fc0b8c9a2b6774f0816635ac6d9afc7/src/main/groovy/org/grails/gradle/plugin/core/GrailsGradlePlugin.groovy#L173-L184

**Comment by matrei** [comment]
Yes, the `org.springframework.boot:spring-boot-dependencies` bom will already be added by https://github.com/grails/grails-gradle-plugin/blob/755b810a1fc0b8c9a2b6774f0816635ac6d9afc7/src/main/groovy/org/grails/gradle/plugin/core/GrailsGradlePlugin.groovy#L189

I don't know if this causes a problem, and if it does, I don't see how we can prevent it, other than not applying the `SpringBootPlugin`, and then to duplicating what it does ourselves (leaving out the `DependencyManagementPluginAction`).

**Comment by codeconsole** [comment]
@matrei I am imagining we can just eliminate

```
     Plugin dependencyManagementPlugin = project.plugins.findPlugin(DependencyManagementPlugin) 
     if (dependencyManagementPlugin == null) { 
         project.plugins.apply(DependencyManagementPlugin) 
     } 
  
     DependencyManagementExtension dme = project.extensions.findByType(DependencyManagementExtension) 
  
     applyBomImport(dme, project) 
```

and just apply our bom directly.

**Comment by codeconsole** [comment]
it turns out grails-bom is being applied twice so that means the spring-bom is being applied 3 times 😮 

[asset-pipeline-grails-5.0.5.pom](https://repo1.maven.org/maven2/com/bertramlabs/plugins/asset-pipeline-grails/5.0.5/asset-pipeline-grails-5.0.5.pom)
```xml
<dependencyManagement>
<dependencies>
  <dependency>
      <groupId>org.grails</groupId>
      <artifactId>grails-bom</artifactId>
      <version>7.0.0-SNAPSHOT</version>
      <scope>import</scope>
      <type>pom</type>
  </dependency>
  <dependency>
      <groupId>org.grails</groupId>
      <artifactId>grails-bom</artifactId>
      <version>7.0.0-SNAPSHOT</version>
      <scope>import</scope>
      <type>pom</type>
  </dependency>
  <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-dependencies</artifactId>
      <version>3.3.5</version>
      <scope>import</scope>
      <type>pom</type>
  </dependency>
</dependencies>
</dependencyManagement>
```

**Comment by koww** [comment]
As a user I would like to ask if we can configure the spring boot version somewhere. A fixed springboot bom is causing trouble for us to update the springboot patch versions for security updates. Besides, we would like to manage the versions of some of the dependencies included by the bom, and this dependencyManagement part is making it very difficult as well.

**Comment by matrei** [comment]
Hi @koww 
Which version of Grails are you using? Also, what approach are you currently trying to override versions from the BOM, that is not succeeding?

Does it not work to override the versions from the BOM with project properties?
```groovy
ext['spring-boot.version'] = '3.4.1'
```

**Comment by jamesfredley** [comment]
Thanks for the report. Closing as fixed by the native Gradle platform integration and single-Grails-BOM enforcement in `ca495ab5` and `6ef4ec20`. Current functional coverage verifies Spring Dependency Management is not applied and a manually declared Grails BOM is not duplicated. Later Spring Boot override behavior is separate scope.
